### PR TITLE
[Reviewer: Ellie] Add function to convert binary to hex

### PR DESCRIPTION
**Pull request by rkd-msw**
This moves it out of Homestead so Sprout can use it.

---
This change is [Reviewable](https://reviewable.io/reviews/metaswitch/cpp-common/564)

### CODE REVIEW

File: src/utils.cpp [hunk 1]
```diff
@@ -269,6 +269,22 @@ void Utils::create_random_token(size_t length,       //< Number of characters.
...
+  result.reserve(2 * len);
+  for (size_t ii = 0; ii < len; ++ii)
+  {
+    const uint8_t b = data[ii];
+    result.push_back(hex_lookup[b >> 4]);
+    result.push_back(hex_lookup[b & 0x0f]);
+  }
+  return result;
+}
+
```
Comment (eleanor-merry) [inline src/utils.cpp]: Excessive lines

**Comment by eleanor-merry** [comment]
LGTM. Are there are UTs that need to move to cpp-common-test?

**Comment by rkd-msw** [comment]
No, this was a private method in Homestead so it wasn't UTed directly, only indirectly through UTs on the other methods.

**Comment by rkd-msw** [comment]
(And it's still covered in UT indirectly, in both Sprout and Homestead now)

**Pull request merged**